vscode/webviews/chat/Transcript.tsx

```diff
@@ -175,57 +209,101 @@ export const Transcript: FC<TranscriptProps> = props => {
+                className={clsx(
+                    'tw-flex tw-flex-col tw-flex-1 tw-overflow-y-auto tw-relative',
+                    {
+                        'tw-flex-grow': transcript.length > 0,
+                    }
+                )}
                 data-scrollable="true"
+                style={fadeStyles}
             >
                 <LastEditorContext.Provider value={lastHumanEditorRef}>
-                    {interactions.map((interaction, i) => (
+                    {/* Show welcome content top when starting a new conversation */}
+                    {shouldShowWelcomeContent && welcomeContentTop}
+                    {chatInteractions.map((interaction, i) => (
                         <TranscriptInteraction
                             key={interaction.humanMessage.index}
-                            activeChatContext={activeChatContext}
-                            setActiveChatContext={setActiveChatContext}
-                            tokenUsage={tokenUsage}
-                            models={models}
-                            chatEnabled={chatEnabled}
-                            userInfo={userInfo}
+                            {...commonTranscriptProps}
                             interaction={interaction}
-                            guardrails={guardrails}
-                            postMessage={postMessage}
-                            copyButtonOnSubmit={copyButtonOnSubmit}
-                            insertButtonOnSubmit={insertButtonOnSubmit}
                             isFirstInteraction={i === 0}
-                            isLastInteraction={i === interactions.length - 1}
+                            isLastInteraction={!inputInteraction && i === chatInteractions.length - 1}
                             isLastSentInteraction={
-                                i === interactions.length - 2 && interaction.assistantMessage !== null
+                                i === chatInteractions.length - 1 &&
+                                interaction.assistantMessage !== null
                             }
                             priorAssistantMessageIsLoading={Boolean(
-                                messageInProgress && interactions.at(i - 1)?.assistantMessage?.isLoading
+                                messageInProgress &&
+                                    chatInteractions.at(i - 1)?.assistantMessage?.isLoading
+                            )}
+                            editorRef={undefined} // Input editor is handled separately
+                        />
+                    ))}
+                    {/* Show loading dots after the last message in scrollable area */}
+                    {messageInProgress && chatInteractions.length > 0 && <LoadingDots />}
+                    {/* Show welcome content bottom when starting a new conversation */}
+                    {shouldShowWelcomeContent && welcomeContentBottom}
+                </LastEditorContext.Provider>
+            </div>
+
```

Comment:
```suggestion
            <ScrollbarMarkers />
```

we should probablay move the marker above the input box